storagetransfer/api/StorageTransfer.Samples.Tests/StorageFixture.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using Google.Apis.Storage.v1.Data;
+using Google.Cloud.Storage.V1;
+using Google.Cloud.StorageTransfer.V1;
+using Xunit;
+namespace StorageTransfer.Samples.Tests
+{
+    [CollectionDefinition(nameof(StorageFixture))]
+    public class StorageFixture : IDisposable, ICollectionFixture<StorageFixture>
+    {
+        public string ProjectId { get; }
+        public string BucketNameSource { get; } = Guid.NewGuid().ToString();
+        public string BucketNameSink { get; } = Guid.NewGuid().ToString();
+        public StorageClient Storage { get; } = StorageClient.Create();
+        public StorageTransferServiceClient Sts { get; } = StorageTransferServiceClient.Create();
+
+        public StorageFixture()
+        {
+            ProjectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+            if (string.IsNullOrWhiteSpace(ProjectId))
+            {
+                throw new Exception("You need to set the Environment variable 'GOOGLE_PROJECT_ID' with your Google Cloud Project's project id.");
+            }
+
+            CreateBucketAndGrantStsPermissions(BucketNameSink);
+            CreateBucketAndGrantStsPermissions(BucketNameSource);
+        }
+
+        private void CreateBucketAndGrantStsPermissions(string bucketName)
```

Comment:
I wonder if some of this code should be part of the sample?
Or are these bucket requirements explained somewhere else?

Reply:
There's docs that explain how to do these permissions with code, but not with a sample. I think there should be code samples, it's just a weird spot because it's something you do with the Storage client, not the Transfer Service client, so there's a question of how do we incorporate a sample for one client in the docs of a completely different client. It's something that I want to figure out and propose down the line, but not at this stage

Reply:
Acked!